packages/next-app/src/app/internalmanager/(components)/TokenSelect.tsx

```diff
@@ -156,15 +172,100 @@ function TokenModal({
     }
   }
 
+  const tokenExplorerUrl = buildExplorerTokenURL({
+    chainId: chain!.id,
+    tokenAddress: tokenSearch.toLowerCase() as `0x${string}`,
+  });
+
   useEffect(() => {
     if (!internalBalanceData?.user?.userInternalBalances) return;
     getWalletBalance(tokenAdresses as `0x${string}`[]);
   }, [internalBalanceData]);
 
+  useEffect(() => {
+    if (addressRegex.test(tokenSearch)) {
+      setIsTokenSearchDisabled(false);
+      if (
+        tokens.filter((token) => filterTokenInput({ tokenSearch, token }))
+          .length < 1
+      ) {
+        fetchSingleTokenBalance({
+          tokenAddress: tokenSearch.toLowerCase() as `0x${string}`,
+        }).then((tokenData) => {
+          setTokens((prev) => [
+            ...prev,
+            {
+              ...tokenData,
+              name: tokenData.symbol,
+              tokenAddress: tokenSearch.toLowerCase() as `0x${string}`,
```

Comment:
not sure I follow -- wouldn't these be have been populated already? Also any reason why we're using `name` instead of the more standard `symbol` for displaying the token nicely on the UI?

Reply:
this is the case for an inputted token that doesn't show on the token list and the user only has this token on their wallet. Therefore, we don't have this data already

`name` is not provided from the wagmi `fetchBalance` so we are using `symbol ` to not do an extra fetch jsut for the name. But the tokens from the InternalBalance and token list have their names showing